### DISCUSSION TITLE
How can I change the theme globally and using native code?

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#5860] [Q&A] [answered]
I want to implement a theme toggle feature.

I have read: https://github.com/slint-ui/slint/discussions/5213
But it only work in that window, the other windows remain the same theme.
Since Palette is a Global Singleton, I think this is a issue.

Also, I would like to ask for help. Is there any way to change the theme globally and switch the theme using native code (Rust)?

**Comment by ogoffart** [comment, marked as answer]
currently, global are global per instances. 
As you've noticed, they are not shared.

So you'll have to make a function in native code that loop over all your window to then call a function in them to change the Palette.color-scheme

**Reply by Horbin-Magician** [reply]
Thank you for your reply.

**Comment by alexeyabel** [comment]
>...then call a function in them to change the Palette.color-scheme

@ogoffart , how exactly would that work? I want to set dark mode for the primary window. AFAIK, I can invoke callbacks from Rust code, but I cannot invoke functions and only functions have a body where I could do `Palette.color-scheme = ColorScheme.dark`. What am I missing?

**Reply by ogoffart** [reply]
You can invoke public functions from native code with invoke_

**Reply by alexeyabel** [reply]
I must have missed that. Reading the documentation I got the wrong impression. Thanks for the clarification, Olivier.
